src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,559 @@
+        Map<String, Object> mapBasedInstallKernel = AccessController.doPrivileged(new PrivilegedExceptionAction<Map<String, Object>>() {
+            @SuppressWarnings({ "unchecked", "resource" })
+            @Override
+            public Map<String, Object> run() throws Exception {
+                ClassLoader loader = new URLClassLoader(new URL[] { installJarFile.toURI().toURL() }, null);
+                Class<Map<String, Object>> clazz;
+                clazz = (Class<Map<String, Object>>) loader.loadClass("com.ibm.ws.install.map.InstallMap");
+                return clazz.newInstance();
+            }
+        });
+        if (mapBasedInstallKernel == null){
+            debug("mbik is null");
+        }
+
+        // Init
+        mapBasedInstallKernel.put("runtime.install.dir", installDirectory);
+        mapBasedInstallKernel.put("install.map.jar", installJarFileSubpath);
+        debug("install.map.jar: " + installJarFileSubpath);
+        debug("install.kernel.init.code: " + mapBasedInstallKernel.get("install.kernel.init.code"));
+        debug("install.kernel.init.error.message: " + mapBasedInstallKernel.get("install.kernel.init.error.message"));
+        File usrDir = new File(installDirectory, "usr/tmp");
+        mapBasedInstallKernel.put("target.user.directory", usrDir);
+        return mapBasedInstallKernel;
+    }
+    
+    /**
+     * Find latest install map jar from lib directory
+     * 
+     * @return the install map jar file
+     */
+    private File getMapBasedInstallKernelJar(File installDirectory) {
+        final String installMapPrefix = "com.ibm.ws.install.map";
+        final String installMapSuffix = ".jar";
+
+        File dir = new File(installDirectory, "lib");
+
+        File[] installMapJars = dir.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.startsWith(installMapPrefix) && name.endsWith(installMapSuffix);
+            }
+        });
+
+        File latestJar = null;
+        if (installMapJars != null && installMapJars.length > 0) {
+            for (File jar : installMapJars) {
+                debug("latestJar: " + latestJar);
+                debug("jar: " + jar);
+
+                if (latestJar == null) {
+                    // first jar found
+                    latestJar = jar;
+                    continue;
+                }
+                String latestJarVersion = extractVersion(latestJar.getName(), installMapPrefix, installMapSuffix);
+                if (latestJarVersion == null) {
+                    // jar without version is the oldest jar 
+                    latestJar = jar;
+                    continue;
+                }
+                String jarVersion = extractVersion(jar.getName(), installMapPrefix, installMapSuffix);
+                if (jarVersion != null && jarVersion.compareTo(latestJarVersion) > 0) {
```

Comment:
I don't think that `compareTo` will work either on version comparisons (I'm not sure how the versioning works for these jars though). For example, `"1.10".compareTo("1.9") > 0` will return `false` even though 1.10 is a later version than 1.9. See the previous comment.